pkgs/by-name/fr/friction-graphics/package.nix

```
@@ -0,0 +1,92 @@
+{ lib
+, clangStdenv
+, cmake
+, expat
+, fetchgit
+, ffmpeg_4-full
+, fontconfig
+, freetype
+, glib
+, gn
+, harfbuzzFull
+, icu
+, libGL
+, libjpeg_turbo
+, libpng
+, libsForQt5
+, libunwind
+, libwebp
+, ninja
+, pcre2
+, pkg-config
+, python3
+, qt5Full
+, xorg
+, zlib
+}:
+
+clangStdenv.mkDerivation rec {
+  pname = "friction.graphics";
+  version = "0.9.6";
+  src = fetchgit {
+    url = "https://github.com/friction2d/friction";
+    rev = version;
+    sha256 = "sha256-QjwkijsTAEEN1VqGtIfsNsHTM1DIRUp+hfABuXoj+nU=";
+    fetchSubmodules = true;
+  };
+  buildInputs = [
```

```suggestion
  version = "0.9.6";

  src = fetchgit {
    url = "https://github.com/friction2d/friction";
    rev = version;
    sha256 = "sha256-QjwkijsTAEEN1VqGtIfsNsHTM1DIRUp+hfABuXoj+nU=";
    fetchSubmodules = true;
  };

  buildInputs = [
```

Please use fetchFromGitHub instead. If possible that uses an archive download which is faster and more reproducible.

---

I tried to, but this didn't fetch submodules. Is it possible to get that to work with submodules?

---

ack, PEBKAC, I had a typo when I was trying to get submodules to work with fetchFromGitHub. Thanks for the prompt